### ISSUE TITLE
IndexError: pop index out of range when instanciating Topology with a list of GeoDataFrame

### ISSUE THREAD

**Issue by aspyk** [mattijn/topojson#203]
Hi !
I just get an `IndexError: pop index out of range` when trying to create a Topology like this:
```
groups = [list(i) for i in zip(*gdf.groupby("datetime"))]
tp.Topology(data=groups[1], object_name=groups[0])
```
It works fine with two objects in the list (as in this [test](https://github.com/mattijn/topojson/blob/263c519a6f4a1df586eea5d96e7798e8ba0986ec/tests/test_extract.py#L481C5-L481C47)) but it seems to break for longer list. The issue comes from [here](https://github.com/mattijn/topojson/blob/263c519a6f4a1df586eea5d96e7798e8ba0986ec/topojson/core/extract.py#L597C23-L597C23):
```
for ix in range(1, len(geom)):
    geom[0].update(geom.pop(ix))
data = geom[0]
```
It breaks because pop reduce the length of the list and so the last indices cannot be reached. Simply using `geom[ix]` instead of `geom.pop(ix)` should fix it. I'm going to test it.

**Comment by mattijn** [comment]
Thanks for raising the issue. If you can submit a PR if you have a fix for it, that would be great. 
Otherwise a simple example how I can reproduce this error would be fine too!
Thanks again for reaching out!

**Comment by mattijn** [comment]
Thanks for raising the issue. If you can submit a PR if you have a fix for it, that would be great. 
Otherwise a simple example how I can reproduce this error would be fine too!
Thanks again for reaching out!

**Comment by aspyk** [comment]
PR done, don't hesitate if something need to be changed :)

**Comment by mattijn** [comment]
Great @aspyk! You are almost there!
You've created a fork and made the changes. 
If it is right there should now appear a green button on the github page of your forked repo of topojson stating, "compare & pull request". 
If you click that button and add a title plus description in the new screen and then click "create pull request", it is official.

**Comment by mattijn** [comment]
Yes, now I see it. Maybe I hadn't refreshed the page.. thanks for the PR! Will check from there

**Comment by aspyk** [comment]
I see there is a problem on the CI, do you know what it is ?

**Comment by mattijn** [comment]
fixed by #204
